Source/Core/DolphinQt/MemoryEngine/MemScanner/MemScanWidget.cpp

```diff
@@ -0,0 +1,423 @@
+
+  QHBoxLayout* multiAddButtons_layout = new QHBoxLayout();
+  multiAddButtons_layout->addWidget(m_btnAddSelection);
+  multiAddButtons_layout->addWidget(m_btnAddAll);
+  multiAddButtons_layout->addWidget(m_btnRemoveSelection);
+
+  QVBoxLayout* results_layout = new QVBoxLayout();
+  results_layout->addWidget(m_lblResultCount);
+  results_layout->addWidget(m_tblResulstList);
+  results_layout->addLayout(multiAddButtons_layout);
+
+  QHBoxLayout* buttons_layout = new QHBoxLayout();
+  buttons_layout->addWidget(m_btnFirstScan);
+  buttons_layout->addWidget(m_btnNextScan);
+  buttons_layout->addWidget(m_btnResetScan);
+
+  QHBoxLayout* searchTerm2_layout = new QHBoxLayout();
+  searchTerm2_layout->setContentsMargins(0, 0, 0, 0);
+  searchTerm2_layout->addWidget(lblAnd);
+  searchTerm2_layout->addWidget(m_txbSearchTerm2);
+  m_searchTerm2Widget->setLayout(searchTerm2_layout);
+
+  QHBoxLayout* searchTerms_layout = new QHBoxLayout();
+  searchTerms_layout->addWidget(m_txbSearchTerm1);
+  searchTerms_layout->addWidget(m_searchTerm2Widget);
+  searchTerms_layout->setSizeConstraint(QLayout::SetMinimumSize);
+
+  QHBoxLayout* layout_buttonsBase = new QHBoxLayout();
+  layout_buttonsBase->addWidget(m_rdbBaseDecimal);
+  layout_buttonsBase->addWidget(m_rdbBaseHexadecimal);
+  layout_buttonsBase->addWidget(m_rdbBaseOctal);
+  layout_buttonsBase->addWidget(m_rdbBaseBinary);
+  m_groupScanBase->setLayout(layout_buttonsBase);
+
+  QHBoxLayout* layout_extraParams = new QHBoxLayout();
+  layout_extraParams->addWidget(m_chkEnforceMemAlignement);
+  layout_extraParams->addWidget(m_chkSignedScan);
+
+  QVBoxLayout* scannerParams_layout = new QVBoxLayout();
+  scannerParams_layout->addLayout(buttons_layout);
+  scannerParams_layout->addWidget(m_cmbScanType);
+  scannerParams_layout->addWidget(m_cmbScanFilter);
+  scannerParams_layout->addLayout(searchTerms_layout);
+  scannerParams_layout->addWidget(m_groupScanBase);
+  scannerParams_layout->addLayout(layout_extraParams);
+  scannerParams_layout->addStretch();
+  scannerParams_layout->setContentsMargins(0, 0, 0, 0);
+
+  QWidget* scannerParamsWidget = new QWidget();
+  scannerParamsWidget->setLayout(scannerParams_layout);
+  scannerParamsWidget->setMinimumWidth(425);
+
+  QHBoxLayout* main_layout = new QHBoxLayout();
+  main_layout->addLayout(results_layout);
+  main_layout->addWidget(scannerParamsWidget);
+  main_layout->setContentsMargins(3, 0, 3, 0);
+
+  setLayout(main_layout);
+}
+
+ResultsListModel* MemScanWidget::getResultListModel() const
+{
+  return m_resultsListModel;
+}
+
+std::vector<u32> MemScanWidget::getAllResults() const
+{
+  return m_memScanner->getResultsConsoleAddr();
+}
+
+QModelIndexList MemScanWidget::getSelectedResults() const
+{
+  return m_tblResulstList->selectionModel()->selectedRows();
+}
+
+MemScanner::ScanFiter MemScanWidget::getSelectedFilter() const
+{
+  int index =
+      GUICommon::g_memScanFilter.indexOf(QRegExp(tr("^") + m_cmbScanFilter->currentText() + tr("$")));
```

Comment:
If we expect this to be a regex, we should likely either make it explicit (and _not_ add the anchors) or at least escape what the user entered before using it as a regex.

Reply:
I am afraid this reaches the limits of my knowledge, so feel free to commit a modification. It so happens that the user can only choose the text that goes here from a list, so we know that there is nothing to escape.